### ISSUE TITLE
Write the Database Accesor Class

### ISSUE THREAD

**Issue by foxtrotzulu94** [foxtrot94/McHacksPokedex#12]
Based on the previous work done, write the corresponding code that handles all the interaction with the packaged database asset. This should include the 1st start copy, methods for general querying and functions to build full Pokemon and associated objects.

**Comment by thineshth** [comment]
Issue-#12-Scrapper/DB Changes

Need to also make sure these changes are all reflected 
- Renamed pokemonID to uniquePokemonID to better represent that it refers to each individual instance of a different pokemon form
- Refactored pokemonID->uniquePokemonID where ever used
- Renamed pokemon table to pokemon_unique_info
- Combined caught, description, hatchTime, catchRate, genderRatioMale attributes from tables into one pokemon_common_info table
- Changed use of uniquePokemonID (formerly pokemonID) in pokemon_eggGroups, pokemon_locations and pokemon_moves, pokemon_caught to use nationalID instead
- Tables Deleted: pokemon_description, pokemon_catchRate, pokemon_genderRatioMale, pokemon_hatchTime

Changed Scrapper to reflect the above changes

**Comment by foxtrotzulu94** [comment]
So I finished putting together most of the system that links UI and backend. I also tried building all full Pokemon Objects in one shot and that is noticeably slow. Parallelizing it is a bit of a mess and doesn't seem to want to work correctly. 

There are a couple of options I see as viable to proceed from here:
- We build Pokemon Objects on-demand. I had been doing this before, but it might imply we lose efficiency to a lot of idle time
- Let the Factory build in the background and load up all Objects. This prevents the user from going beyond the PokedexActivity until the detailedPokemonShortList is available
- Make a mix of both (which is a bit of what is in the latest commit). Problem is we run into many synchronization issues, which we can only solve by placing locks all over the place

Either way, PokedexManager and PokemonFactory might need some refactoring in all cases. Take a look and let me know what you think
